MapboxNavigation/NavigationView.swift

```diff
@@ -0,0 +1,92 @@
+import UIKit
+import MapboxDirections
+
+/**
+ A view that represents the root view of the MapboxNavigation drop-in UI.
+ 
+ ## Components
+ 
+ 1. InstructionsBannerView
+ 2. InformationStackView
+ 3. BottomBannerView
+ 4. ResumeButton
+ 5. WayNameLabel
+ 6. FloatingStackView
+ 7. NavigationMapView
+ 
+ ```
+ +--------------------+
+ |         1          |
+ +--------------------+
+ |         2          |
+ +----------------+---+
+ |                |   |
+ |                | 6 |
+ |                |   |
+ |         7      +---+
+ |                    |
+ |                    |
+ |                    |
+ +------------+       |
+ |  4  ||  5  |       |
+ +------------+-------+
+ |         3          |
+ +--------------------+
+ ```
+*/
+@IBDesignable
+@objc(MBNavigationView)
+open class NavigationView: UIView {
+    
+    weak var mapView: NavigationMapView!
+    weak var wayNameLabel: WayNameLabel!
+    weak var bottomBannerView: BottomBannerView!
+    weak var bottomBannerContentView: BottomBannerContentView!
+    weak var instructionsBannerView: InstructionsBannerView!
+    weak var instructionsBannerContentView: InstructionsBannerContentView!
+    weak var lanesView: LanesView!
+    weak var nextBannerView: NextBannerView!
+    weak var statusView: StatusView!
+    weak var resumeButton: ResumeButton!
+    // Vertically laid-out stack view below the instructions banner consisting of StatusView, NextBannerView, and LanesView.
+    weak var informationStackView: UIStackView!
+    // Vertically laid-out stack view below the information stack view ontop of the map view, docked
+    // to the top right, consisting of Overview, Mute, and Report button.
+    weak var floatingStackView: UIStackView!
```

Comment:
The one major comment I have on this PR is this: Since this is no longer IB-based, it would be much cleaner to use a `lazy var` to set up all of these child views, like so:

```swift
lazy var floatingStackView: UIStackView = { 
    let stack = UIStackView()
    stack.axs = .vertical
    stack.distribution = .equalSpacing
    stack.spacing = 8
    stack.translatesAutoresizingMaskIntoConstraints = false
    return stack
} ()
```

then, `setupViews()` becomes this:

```swift
func setupViews() {
    [..., floatingStackView, overviewButton, muteButton, reportButton, ...].forEach(addSubview(_:))
}
```

However, this does work for now, so if we want to just get this across the line then I'll happily take care of it in my PR.

Reply:
Totally agree about the lazy-loaded ivars. Even more if we decide to lazy-load localized strings. However, I'm going to merge this as is b/c the utilization of this class will be done in a subsequent PR which is when we can measure the impact.